File: LondonFhirService.Manage.Client/src/models/foundations/patients/StructuredRecordResponse.ts
// The payload and the id the call was filed under. They arrive together - the payload in the
// body, the correlation id in the X-Correlation-Id response header - so they travel together
// rather than the header being read again further up.
export type StructuredRecordResponse = {
    payloadText: string;

    // 32 hex characters, no dashes: the W3C trace id. Empty when the API did not send the header,
    // which is what an older build of the host does, so the page must cope with not having it.
    correlationId: string;
};

File: LondonFhirService.Manage/Models/Foundations/Patients/StructuredRecordResponse.cs
// ---------------------------------------------------------
// Copyright (c) North East London ICB. All rights reserved.
// ---------------------------------------------------------

namespace LondonFhirService.Manage.Models.Foundations.Patients
{
    /// <summary>
    /// What the provider answered, and the id the Api host filed the call under.
    ///
    /// The two travel together because they arrive together and are separated immediately after:
    /// the controller writes the id to a response header and the payload to the body. Keeping
    /// them paired this far means the controller never has to ask the broker a second question
    /// about a response that no longer exists.
    ///
    /// CorrelationId is empty when the upstream sent no header, which is what an older build of
    /// the Api does, so the page has to cope with not having one.
    /// </summary>
    public class StructuredRecordResponse
    {
        public string PayloadText { get; set; }
        public string CorrelationId { get; set; }
    }
}
